### PR TITLE
[ASLayoutSpec] Make child and children a property

### PR DESCRIPTION
**Pull request by levi**
Nice dot syntax in obj-c!

### CODE REVIEW

**Comment by maicki** [comment]
I agree with this change and it should be a property, I think we just have to be careful about this changes as in Swift this will change from accessing `children` as function: `node.children()` to property access: `node.children`

**Comment by maicki** [comment]
Related: http://indiestack.com/2016/06/swifty-sdk-changes/

**Comment by Adlai-Holler** [comment]
The documentation comments need some phrasing changes to match the new property-ness. Otherwise 👍

File: AsyncDisplayKit/Layout/ASLayoutSpec.h [hunk 1]
```diff
@@ -45,7 +45,7 @@ NS_ASSUME_NONNULL_BEGIN
  * setChild:forIdentifier: internally. For example, ASBackgroundLayoutSpec exposes a backgroundChild
```
Comment (appleguy) [inline AsyncDisplayKit/Layout/ASLayoutSpec.h]: Does the comment need to mention anything that this does internally?  Seems odd.

**Comment by appleguy** [comment]
@levi could you tweak the comments as appropriate?  @Adlai-Holler what steps do we need to make this Swift-safe?  Honestly I doubt many people are reading the children value, so hopefully it is safe to change--or does it affect writing too?

**Comment by maicki** [comment]
@appleguy It will affect writing too!

The change would be on the Swift side:
*Reading:*
`node.children()` to `node.children`

*Writing:*
`node.setChildren([child])` to `node.children = [child]`

**Comment by maicki** [comment]
Hey @levi can you please rebase the master branch of your fork on top of the latest upstream master and rebase this PR branch on top. We will likely merge that pretty soon. Thanks!

**Comment by levi** [comment]
@maicki not able to do this soon, could you commandeer this diff?

**Comment by maicki** [comment]
Closing this PR and moving it over to: #2041

**Pull request closed without merging**